### PR TITLE
docs(Callback): add solutions for Callback chapter.

### PR DESCRIPTION
**Pull request by cRAN-cg**
Solutions for exercise
>Side Effects With forEach
>Annotating Data

### CODE REVIEW

**Comment by cRAN-cg** [comment]
@gvwilson could you please review

**Comment by gvwilson** [comment]
HI @MayaGans - can you please take a look at this one and give Chiranjeev some feedback? Thanks - G

File: docs/answers.md [hunk 1]
```diff
@@ -5,6 +5,39 @@ title: Answers
 
 ---
 
+## Callbacks
+
+### Side Effects with **forEach**
+
+```{js}
```
Comment (MayaGans) [inline docs/answers.md]: This looks great!

File: docs/answers.md [hunk 2]
```diff
@@ -5,6 +5,39 @@ title: Answers
...
+const doubleInPlace = (array) => {
+  array.forEach((value, location, container) => {
+    container[location] = value * 2;
+  })
+}
+```
```
Comment (MayaGans) [inline docs/answers.md]: This looks great!

File: docs/answers.md [hunk 3]
```diff
@@ -5,6 +5,39 @@ title: Answers
...
+            'sex': currentValue.sex,
+            'species': currentValue.species
+        })
+    }
+    return accumulator
+}
+
+const annotateData = (array) => array.reduce(reducer, [])
+
+const newData = annotateData(data)
```
Comment (MayaGans) [inline docs/answers.md]: This works, but using `.map` and `.filter` may be a more elegant solution? 

```
const cleanData = (d) => {
    const out = d.map((item, idx) => {
        return {
            seq: idx + 1,
            year: item.date.slice(0, 4),
            sex: item.sex,
            species: item.species
        }
    })
    .filter(x => x.sex === 'F')
    return out
}
```

Comment (MayaGans) [reply docs/answers.md]: @cRAN-cg - this is more a suggestion than an issue that needs to be resolved, let me know what you think.

**Comment by cRAN-cg** [comment]
Yes, I totally agree with you. I had a misconception of using `Array.reduce`for this solution as it was mentioned 

> You will probably want to use Array.reduce to generate the sequence numbers.

We can include in both the solutions, explicitly indicating that the `.map` , `.filter` way is a more glorious way to do so.
What do you think about it @MayaGans

File: docs/answers.md (hunk 3 above)
Comment (cRAN-cg) [reply docs/answers.md]: Yes, I totally agree with you. I had a misconception of using Array.reducefor this solution as it was mentioned

    You will probably want to use Array.reduce to generate the sequence numbers.

We can include in both the solutions, explicitly indicating that the .map , .filter way is a more glorious way to do so.
What do you think about it @MayaGans

Comment (MayaGans) [reply docs/answers.md]: Oh good catch - not a misconception at all, you're totally right. I'm going to merge as I think it'd be a little confusing to the reader if we present multiple explanations. Thanks!

Comment (cRAN-cg) [reply docs/answers.md]: thanks :))

**Pull request merged**